app/src/main/java/org/mozilla/fenix/collections/CollectionCreationUIView.kt

```diff
@@ -86,80 +106,104 @@ class CollectionCreationUIView(
 
                 back_button.text = view.context.getString(R.string.create_collection_select_tabs)
 
-                val buttonText = if (it.selectedTabs.isEmpty()) {
+                val selectTabsText = if (it.selectedTabs.isEmpty()) {
                     view.context.getString(R.string.create_collection_save_to_collection_empty)
                 } else {
                     view.context.getString(
                         R.string.create_collection_save_to_collection_full,
                         it.selectedTabs.size
                     )
                 }
+                view.select_tabs_layout_text.text = selectTabsText
 
-                tab_list.visibility = View.VISIBLE
-                select_all_button.visibility = View.VISIBLE
-                add_tabs_button.visibility = View.VISIBLE
-
-                val enableSaveButton = it.selectedTabs.isNotEmpty()
-                view.add_tabs_button.isClickable = enableSaveButton
-
-                view.add_tabs_button.contentDescription = buttonText
-                view.add_tabs_button_text.text = buttonText
+                save_button.visibility = if (it.selectedTabs.isEmpty()) {
+                    View.GONE
+                } else {
+                    View.VISIBLE
+                }
             }
             is SaveCollectionStep.SelectCollection -> {
+                // Only show selected tabs and hide checkboxes
+                collectionCreationTabListAdapter.updateData(it.selectedTabs.toList(), setOf(), true)
+
                 back_button.setOnClickListener {
                     actionEmitter.onNext(CollectionCreationAction.BackPressed(SaveCollectionStep.SelectCollection))
                 }
-                collections_list.visibility = View.VISIBLE
-                add_collection_button.visibility = View.VISIBLE
-                divider.visibility = View.VISIBLE
-                tab_list.visibility = View.GONE
-                select_all_button.visibility = View.GONE
-                add_tabs_button.visibility = View.GONE
-                name_collection_edittext.visibility = View.GONE
-
+                TransitionManager.beginDelayedTransition(
+                    view.collection_constraint_layout,
+                    transition
+                )
+                val constraint = selectCollectionConstraints
+                constraint.applyTo(view.collection_constraint_layout)
                 back_button.text =
                     view.context.getString(R.string.create_collection_select_collection)
             }
             is SaveCollectionStep.NameCollection -> {
                 back_button.setOnClickListener {
-                    actionEmitter.onNext(CollectionCreationAction.BackPressed(SaveCollectionStep.NameCollection))
+                    name_collection_edittext.hideKeyboard()
+                    val handler = Handler()
+                    handler.postDelayed({
```

Comment:
Do we need a 200ms delay or will this work with just name_collection_edittext.post( Runnable { } )?

Reply:
I can show you tomorrow, but if the keyboard isn't hidden before the transition starts, the ConstraintLayout gets laid out above a phantom keyboard. 👀 I basically want to ensure the keyboard has had time to close before emitting the action to change the scene, and this is the best way I could figure out to do it since there's not a keyboard listener.

Reply:
LOL, do I ever know there isn't a keyboard listener!. I suppose we can use this workaround for now.